odl/discr/discr_ops.py

```diff
@@ -130,110 +144,212 @@ def finite_diff(f, out=None, axis=0, dx=1.0, edge_order=2,
 
     method, method_in = str(method).lower(), method
     if method not in ('central', 'forward', 'backward'):
-        raise ValueError("method '{}' has to be central, forward or backward"
+        raise ValueError('method {} is not central, forward or backward'
                          ''.format(method_in))
 
+    if isinstance(padding, (int, float)) or padding in (None, 'edge'):
+        if isinstance(padding, int):
+            padding = float(padding)
+    else:
+        raise ValueError('padding value ({}) not valid'.format(padding))
+
+    if edge_order is None:
+        if method == 'central':
+            edge_order = 2
+        else:
+            edge_order = 1
+    else:
+        if edge_order not in [1, 2]:
+            raise ValueError('edge order ({0}) not valid'.format(edge_order))
+
+    if f_arr.shape[axis] == 2 and edge_order == 2:
+        raise ValueError('shape ({0}) of array to small to use one-sided '
+                         'second-order accurate edges'.format(f_arr.shape))
+
+    if padding is not None:
+        if method == 'central' and edge_order == 1:
+            raise ValueError(
+                'central differences with padding cannot be used with '
+                'first-order accurate edges')
```

Comment:
Message OK, but write something in the docs, too. This becomes quite much (too much `Parameters` section), so I suggest that you make a section `Notes` where you describe all the combinations and limitations.